src/Maestro/maestro-angular/src/maestro-client/models.ts

```diff
@@ -1007,6 +1008,238 @@ export class BuildRef {
     }
 }
 
+export class FlowGraph {
+    public constructor(
+        {
+            nodes,
+            edges,
+        }: {
+            nodes: FlowRef[],
+            edges: FlowEdge[],
+        }
+    ) {
+        this._nodes = nodes;
+        this._edges = edges;
+    }
+
+    private _nodes: FlowRef[];
+    private _edges: FlowEdge[];
+
+    public get nodes(): FlowRef[] {
+        return this._nodes;
+    }
+
+    public get edges(): FlowEdge[] {
+        return this._edges;
+    }
+    
+    public isValid(): boolean {
+        return (
+            this._nodes !== undefined && this._edges !== undefined
+        );
+    }
+
+    public static fromRawObject(value: any): FlowGraph {
+        let result = new FlowGraph({
+            nodes: value["flowRefs"] == null ? undefined : value["flowRefs"].map((n: any) => FlowRef.fromRawObject(n)) as any,
+            edges: value["flowEdges"] == null ? undefined : value["flowEdges"].map((e: any) => FlowEdge.fromRawObject(e)) as any,
+        });
+        return result;
+    }
+
+    public static toRawObject(value: FlowGraph): any {
+        let result: any = {};
+        result["flowRef"] = value._nodes.map((n: any) => FlowRef.toRawObject(n));
+        result["flowEdge"] = value._edges.map((e: any) => FlowEdge.toRawObject(e));
+        return result;
+    }
+}
+
+export class FlowRef {
+    public constructor(
+        {
+            id,
+            repository,
```

Comment:
This is a really weird signature.  We should extract an interface for this struct of 9 things, and just have `public constructor data: WhateverData` and then `this._id = data.id`